pangeo_forge_recipes/storage.py

```diff
@@ -240,3 +242,15 @@ def _slugify(value: str) -> str:
     value = unicodedata.normalize("NFKD", value).encode("ascii", "ignore").decode("ascii")
     value = re.sub(r"[^.\w\s-]+", "_", value.lower())
     return re.sub(r"[-\s]+", "-", value).strip("-_")
+
+
+def _add_query_string_secrets(fname: str, secrets: str) -> str:
+    parsed = urlparse(fname)
+    query = secrets if len(parsed.query) == 0 else f"{parsed.query}&{secrets}"
+    parsed = parsed._replace(query=query)
```

Comment:
Great point which I hadn't looked into before pushing, though strangely it seems this *is* the recommended API for named tuples?

"To prevent conflicts with field names, the method and attribute names start with an underscore."
https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace

"As is the case with all named tuples, the subclass has a few additional methods and attributes that are particularly useful. One such method is `_replace()`. The `_replace()` method will return a new ParseResult object replacing specified fields with new values.'
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

Reviewing these docs turned up [`parse_qs`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs) and [`urlencode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode), which will allow users to pass query string params as a dictionary. I'll include this upgrade in the forthcoming review-ready revision.